### PR TITLE
added two buttons and wrote specs

### PR DESCRIPTION
**Pull request by FaxriddinMaxmadiyorov**
@XaoGao hello, it is complete, could you please check?

### CODE REVIEW

File: app/controllers/vacancies_controller.rb [hunk 1]
```diff
@@ -47,6 +47,14 @@ def destroy
     redirect_to vacancies_path
   end
 
+  def my_vacancies
+    if current_user.nil?
```
Comment (XaoGao) [inline app/controllers/vacancies_controller.rb:51]: for this we use method's devise gem, before_action :authenticate_user!, only: []

File: app/views/layouts/_header.html.slim [hunk 2]
```diff
@@ -8,6 +8,11 @@ nav.navbar.navbar-expand-lg.bg-body-tertiary
       ul.navbar-nav.me-auto.mb-2.mb-lg-0
         li.nav-item
           = link_to "Home", root_path, class: "nav-link fs-4 text-black"
+        li.nav-item
+          = link_to "Vacancies", vacancies_path, class: "nav-link fs-4 text-black"
+        - unless current_user.nil?
```
Comment (XaoGao) [inline app/views/layouts/_header.html.slim:13]: use method user_signed_in?

File: spec/requests/vacancies_spec.rb [hunk 3]
```diff
@@ -1,5 +1,20 @@
 require "rails_helper"
 
 RSpec.describe "Vacancies", type: :request do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe "GET /vacancies" do
+    it "returns http success" do
```
Comment (XaoGao) [inline spec/requests/vacancies_spec.rb:5]: All it does some things or return, remove word "return" in spec text

File: spec/requests/vacancies_spec.rb [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+      get vacancies_path
+      expect(response).to have_http_status(:success)
+    end
+
+    it "redirects to root_path if current_user is not authenticated" do
```
Comment (XaoGao) [inline spec/requests/vacancies_spec.rb:10]: redirect to root_path if **user** is not authenticated

File: spec/requests/vacancies_spec.rb [hunk 5]
```diff
@@ -1,5 +1,20 @@
...
+      get my_vacancies_vacancies_path
+      expect(response).to redirect_to(root_path)
+    end
+
+    it "redirects with 302 status code if current_user is not authenticated" do
```
Comment (XaoGao) [inline spec/requests/vacancies_spec.rb:15]: redirect user to root path

File: app/views/vacancies/my_vacancies.html.slim [hunk 6]
```diff
@@ -0,0 +1,13 @@
+.container
+  .row
+    .col.text-center.d-flex.justify-content-center
+      = button_to "New vacancy", new_vacancy_path, method: :get, class: "btn btn-primary mx-2"
+.row.row-cols-1.row-cols-md-2.row-cols-xl-4.g-4.mx-20.mt-0
+  - @my_vacancies.each do |vacancy|
+    .col.d-flex.align-items-stretch
```
Comment (XaoGao) [inline app/views/vacancies/my_vacancies.html.slim:7]: create new partial views "_vacancy.slim.html" which show one vacancy

**Review by XaoGao — COMMENTED** [review]
@FaxriddinMaxmadiyorov pls fix

File: app/views/layouts/_header.html.slim (hunk 2 above)
Comment (FaxriddinMaxmadiyorov) [reply app/views/layouts/_header.html.slim:13]: @XaoGao oh, thank you

File: spec/requests/vacancies_spec.rb (hunk 4 above)
Comment (FaxriddinMaxmadiyorov) [reply spec/requests/vacancies_spec.rb:10]: @XaoGao ok

File: app/controllers/vacancies_controller.rb (hunk 1 above)
Comment (FaxriddinMaxmadiyorov) [reply app/controllers/vacancies_controller.rb:51]: done

**Review by XaoGao — APPROVED** [review]
@FaxriddinMaxmadiyorov approve

**Pull request merged**